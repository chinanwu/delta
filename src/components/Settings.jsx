import React, { useCallback, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import getThemeClassname from '../functions/getThemeClassname';

import { applyTheme } from '../thunk/ThemeThunk.jsx';

import './Settings.less';

export const Settings = ({ dark, onToggle }) => {
	useEffect(() => {
		document.title = 'Settings - Mairead';
	}, []);

	const handleChange = useCallback(
		event => {
			onToggle(event.target.checked);
		},
		[onToggle]
	);

	return (
		<div className={getThemeClassname('Settings', dark)}>
			<div className="Settings__nav" role="navigation">
				<Link className={getThemeClassname('Settings__navBtn', dark)} to="/">
					Home
				</Link>
				<Link
					className={getThemeClassname('Settings__navBtn', dark)}
					to="/about"
				>
					About
				</Link>
				<Link
					className={getThemeClassname('Settings__navBtn--disabled', dark)}
					to="/settings"
				>
					Settings
				</Link>
			</div>
			<div className="Settings__header">Settings</div>
			<div className="Settings__options">
				<div className="Settings__option">
					<div className="Settings__label">Dark Theme:</div>
					<label className="Settings_toggle">
						<input
							type="checkbox"
							onChange={handleChange}
							checked={dark}
							aria-checked={dark}
						/>
						<span
							className={getThemeClassname('Settings_toggleSlider', dark)}
						/>
					</label>
				</div>
			</div>
		</div>
	);
};

Settings.propTypes = {
	dark: PropTypes.bool,
};

export const mapStateToProps = ({ theme }) => ({
	dark: theme.dark,
});

export const mapDispatchToProps = {
	onToggle: applyTheme,
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Settings);
