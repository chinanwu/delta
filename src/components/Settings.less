@import '../style/styles';

.Settings {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	height: 100%;
}

.Settings--dark {
	@dark();
}

.Settings__header {
	@header();
	margin-bottom: 24px;
}

.Settings__option {
	display: flex;
	font-size: 24px;
}

.Settings__label {
	margin-right: 12px;
}

.Settings__checkbox {
	width: 24px;
}

.Settings__toggle {
	position: relative;
	display: inline-block;
	width: 60px;
	height: 34px;
}

.Settings__toggle input {
	opacity: 0;
	width: 0;
	height: 0;
}

.Settings__toggleSlider {
	position: absolute;
	cursor: pointer;
	top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	background-color: #ccc;
	-webkit-transition: 0.4s;
	transition: 0.4s;
	border-radius: 34px;
}

.Settings__toggleSlider:before {
	position: absolute;
	content: '';
	height: 26px;
	width: 26px;
	left: 4px;
	bottom: 4px;
	background-color: white;
	-webkit-transition: 0.4s;
	transition: 0.4s;
	border-radius: 50%;
}

input:checked + .Settings__toggleSlider {
	background-color: #2196f3;
}

input:focus + .Settings__toggleSlider {
	box-shadow: 0 0 1px #2196f3;
}

input:checked + .Settings__toggleSlider--dark {
	background-color: @darkOther;
}

input:focus + .Settings__toggleSlider--dark {
	box-shadow: 0 0 1px @darkOther;
}

input:checked + .Settings__toggleSlider:before {
	-webkit-transform: translateX(26px);
	-ms-transform: translateX(26px);
	transform: translateX(26px);
}
