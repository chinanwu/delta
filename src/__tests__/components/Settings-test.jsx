import { shallow } from 'enzyme';
import React from 'react';

import { Settings, mapStateToProps } from '../../components/Settings.jsx';

jest.unmock('../../components/Settings.jsx');

describe('Settings component', () => {
	describe('rendering', () => {
		it('renders Settings', () => {
			const wrapper = shallow(<Settings />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});

	describe('mapStateToProps', () => {
		it('returned mapped properties', () => {
			const theme = { dark: true };
			const state = { theme };
			expect(mapStateToProps(state)).toEqual(theme);
		});
	});
});
