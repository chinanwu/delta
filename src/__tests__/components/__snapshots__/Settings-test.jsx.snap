// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component rendering renders Settings 1`] = `
<div>
  <Memo(Connect(NavBar))
    activeTab="settings"
  />
  <div
    className="Settings__header"
  >
    Settings
  </div>
  <div
    className="Settings__options"
  >
    <div
      className="Settings__option"
    >
      <div
        className="Settings__label"
      >
        Dark Theme:
      </div>
      <label
        className="Settings__toggle"
      >
        <input
          onChange={[Function]}
          type="checkbox"
        />
        <span />
      </label>
    </div>
  </div>
</div>
`;
